app/services/EtmpRegimeService.scala

```diff
@@ -114,13 +117,20 @@ class EtmpRegimeService @Inject()(etmpConnector: EtmpConnector,
     }
   }
 
+  def compareOptionalStrings(bcdValue: Option[String], erdValue: Option[String]): Boolean = {
+    if(bcdValue.isEmpty && erdValue.isEmpty) {
+      true
+    } else {
+      bcdValue.map(_.toUpperCase).contains(erdValue.map(_.toUpperCase).getOrElse(""))
```

Comment:
bcdValue.isEmpty && erdValue.isEmpty ||
  bcdValue.map(_.toUpperCase).contains(erdValue.map(_.toUpperCase).getOrElse("")

Reply:
I don't think this is an OR case. If we have 2 Nones, the right hand side of the or statement will pass and we don't want it to.

Reply:
The left hand statement short circuits the right hand if it's true (2 Nones)

Reply:
Oh wait, you're right.